apps/authentication/views.py

```diff
@@ -83,9 +85,12 @@ def register(request):
 
                 # Create the registration token
                 token = uuid.uuid4().hex
-                rt = RegisterToken(user=user, email=email.email, token=token)
-                rt.save()
-
+                try:
+                    rt = RegisterToken(user=user, email=email.email, token=token)
+                    rt.save()
+                    log.info('Successfully registered token for %s' % request.user)
+                except:
+                    log.error('Failed to register token for %s' % request.user)
```

Comment:
I did? Can't remember, anyway, the except should not be without an explicit exception. E.g `except IntegrityError` or similar. And it can certainly fail, as all things can. But with the new 500 temlate, it should log interesting information and then re-raise the exception to render the template.